index.js

```diff
@@ -65,29 +67,34 @@ class APCAccess {
   getServices() {
     // Required by Homebridge; expose services this accessory claims to have
     const services = [this.informationService, this.contactSensor, this.batteryService];
-    if (this.config.temperatureSensor === true) {
+
+    if (this.temperatureService) {
       services.push(this.temperatureService);
     }
+
     return services;
   }
 
   getLatestJSON() {
     this.client.getStatusJson().then((result) => {
       this.latestJSON = result;
       this.doPolledChecks();
+    })
+    .catch((err) => {
+      this.log.error("Polling device failed:", err);
     });
   }
 
   getBatteryLevel(callback) {
     // BCHARGE
     let battPctValue = 0;
     const battVal = this.latestJSON.BCHARGE;
-    if (battVal !== undefined) {
+
+    if (battVal) {
```

Comment:
This I'm nervous of, a previous bug report regarding this value's incoming data was a nightmare to get stable.